smart-engine/src/main/java/org/smartdata/server/engine/rule/copy/FileCopyDrPlugin.java

```diff
@@ -39,39 +39,47 @@
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
-import java.util.StringJoiner;
 
 import static org.smartdata.utils.StringUtil.ssmPatternsToRegex;
 
 public class FileCopyDrPlugin implements RuleExecutorPlugin {
   private final MetaStore metaStore;
+  private final FileCopyScheduleStrategy copyScheduleStrategy;
   private final Map<Long, List<BackUpInfo>> backups = new HashMap<>();
   private static final Logger LOG =
       LoggerFactory.getLogger(FileCopyDrPlugin.class.getName());
 
-  public FileCopyDrPlugin(MetaStore metaStore) {
+  public FileCopyDrPlugin(MetaStore metaStore, FileCopyScheduleStrategy copyScheduleStrategy) {
     this.metaStore = metaStore;
+    this.copyScheduleStrategy = copyScheduleStrategy;
   }
 
-  public void onNewRuleExecutor(final RuleInfo ruleInfo, TranslateResult tResult) {
+  public void onNewRuleExecutor(final RuleInfo ruleInfo, RuleTranslationResult tResult) {
```

Comment:
Consider breaking down this large method into smaller ones to improve code comprehension.